Updates to debugFillProperties to test all properties in slider.dart and slider_test.dart

sladecapasso opened the pull request:
## Description

* This PR is to add DiagnosticsProperties tests to Material Slider, to address issue #25671

## Related Issues

* No related issues

## Tests

I added the following tests:

* DiagnosticsProperties

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

<!-- Links -->
[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

sladecapasso committed:
Updates to slider_test.dart

Fixed error that required onChanged.

sladecapasso commented:
@clocksmith I think new this PR should fix the errors, but it looks like the build is currently broken. Wasn't sure if this is an outside issue I need to wait for resolution on or caused by the pull request.

clocksmith commented on packages/flutter/test/material/slider_test.dart:
```diff
@@ -1547,4 +1547,36 @@ void main() {
     expect(renderObject.size.height, 200);
   });
+
+  //DiagnosticsProperties Tests
```
You can probably remove this comment

clocksmith commented on packages/flutter/test/material/slider_test.dart:
```diff
@@ -1547,4 +1547,36 @@ void main() {
+      onChanged: null,
+    ).debugFillProperties(builder);
+
+    print(Slider);
```
remove print

clocksmith commented on packages/flutter/test/material/slider_test.dart:
```diff
@@ -1547,4 +1547,36 @@ void main() {
+      'activeColor: Colors.blue',
+      'inactiveColor: Colors.blue',
+      'label: "Set a value"',
+      'OnChanged: null',
```
should this be onChanged?

clocksmith reviewed (changes requested):
looking good

clocksmith commented:
This will also need to update the current debugFillProperties with the new properties

sladecapasso committed:
Updated slider.dart and slider_test.dart debugFillProperties

Updated slider.dart debugFillProperties to include all properties and slider_test debugFillProperties to test all properties

clocksmith commented on packages/flutter/test/material/slider_test.dart:
```diff
@@ -1547,4 +1547,41 @@ void main() {
+
+    expect(description, <String>[
+      'value: 50.0',
+      'onChanged: disabled',
```
This can just say "disabled"

sladecapasso committed:
Updated Slider onChanged property

sladecapasso committed:
Updated debugFillProperties to ignore testing methods that are null in test

Per feedback from Hans/Anthony on Flutter Material

sladecapasso commented:
Hi @HansMuller , wanted to provide an update here since Anthony is OOO. He let me know that you wanted Slider methods that are not required and set to null should not show anything in the debugFillProperties output, so I committed those changes earlier and am just waiting for the checks to finish running (and flutter-build to not be broken).

Still pretty new at Flutter (and Github), so let me know if there are any other changes you think are necessary.

HansMuller commented on packages/flutter/lib/src/material/slider.dart:
```diff
@@ -381,8 +381,16 @@ class Slider extends StatefulWidget {
     properties.add(DoubleProperty('value', value));
+    properties.add(ObjectFlagProperty<ValueChanged<double>>('onChanged', onChanged, ifNull: 'disabled'));
+    properties.add(ObjectFlagProperty<ValueChanged<double>>.has('onChangeStart', onChangeStart));
+    properties.add(ObjectFlagProperty<ValueChanged<double>>.has('onChangeEnd', onChangeEnd));
```
Nice use of ObjectFlagProperty.has. I didn't know we has that :-)

sladecapasso replied:
Credit goes to @johnsonmh on that one!

HansMuller commented on packages/flutter/test/material/slider_test.dart:
```diff
@@ -1547,4 +1547,38 @@ void main() {
+    ).debugFillProperties(builder);
+
+    final List<String> description = builder.properties
+        .where((DiagnosticsNode node) => !node.isFiltered(DiagnosticLevel.info))
```
two space indent

HansMuller reviewed (approved):
LGTM

sladecapasso committed:
Updated spacing in debugFillProperties

Per feedback from Hans
